### ISSUE TITLE
questions about the pytorch code

### ISSUE THREAD

**Issue by wl082013** [ialhashim/DenseDepth#162]
Hi thanks for the nice code.
I just went through the pytorch train code, I was confused about the data normalization.

 if self.is_test:
            depth = self.to_tensor(depth).float() / 1000
        else:            
            depth = self.to_tensor(depth).float() * 1000
what is the intention of *1000 in the training and /1`000- in the test ?
I saw the ground truth depth was normalized by maxdepth/depth, which I somehow could not understand.
Could you help explain this? 
I am trying to use my own data, and I found this way of depth loading has some problems.

moreover, in the train.py, there are no checkpoint saving and performance evaluation codes. Did you provide in somewhere else?  Thanks for the nice work.

**Comment by tomashlehmann** [comment]
Hi guys, 
I also have a question about the pytorch code. After how many epochs the results should be similar to the results from the paper? After 20th epoch:
a1=0.56, a2=0.83, a3=0.93, rel=0.25, rms=1, log_10=0.11
Did anyone check if the new implementation is correct and if achieved results are as good as the one from publication? 
I will be glad for any answer :)

**Comment by tanmayGIT** [comment]
Hi, 
I see that the authors are completely unresponsive ! Which is bizarre and not helpful. I have the same question for the PyTorch code. I don't actually understand why they did the following ?

if self.is_test:
        depth = self.to_tensor(depth).float() / 1000
else:
        depth = self.to_tensor(depth).float() * 1000 
        else:

But if you notice carefully, in "to_tensor" function, they normalized the values between 0-255 : return img.float().div(255). So the value becomes within 0-1. Then if you multiply any value by 1000 then definitely all the values will be between 0-1000. 

But then also they did this, maybe just to be on the safe side and to be sure that the values are within 0-1000   :
        # put in the expected range
        depth = torch.clamp(depth, 10, 1000)

The possible reason which I can imagine is that the reason for dividing by 1000 is they consider that no distance value will be more than 1000 (considering all the distance matrix of the complete dataset) in any way. So they decide to normalize by 1000. 

But again there is no understandable reason for multiplying by 1000 for training and dividing in the case of testing. 

Another point, if you carefully debug and run, then you will see that the control never enters in "if self.is_test:" that means they never do this operation and they always do multiplication by 1000 and never does the division by 1000.

The flow of code you can check in this order :

getTrainingTestingData(batch_size)    ->
transformed_testing = depthDatasetMemory(data, nyu2_train, transform=getNoTransform())      ->
getNoTransform(is_test=False) -> 
ToTensor(object)  -> 
__init__(self, is_test=False)   ->

        if self.is_test:
            depth = self.to_tensor(depth).float() / 1000



Hence, the depth matrix gets never divided by 1000.

**Comment by ialhashim** [comment]
As explicitly stated in the README. The PyTorch code is experimental, it is not supported and only provided as a courtesy and a start for others to build on.

**Comment by tanmayGIT** [comment]
Thanks for the response. Could you please answer the question i.e. why the values are normalized between 10-1000. Because you did a similar operation in your original code too. 

depth = torch.clamp(depth, 10, 1000)

and it was also not clear why you normalize the predicted test image like this in "/PyTorch/utils.py" 

def DepthNorm(depth, maxDepth=1000.0): 
    return maxDepth / depth

Thanks in advance for your reply.
